### DISCUSSION TITLE
Issue with WPF project single-file publish for win-x64: Unable to run and separate CEFSharp DLLs into a folder

### DISCUSSION THREAD

**Discussion by AB901** [cefsharp/CefSharp#4944] [Q&A]
I am working on a WPF project and trying to generate a single-file executable for win-x64. However, after generating the single file, it fails to run. Additionally, I would like to have the CEFSharp-related DLLs placed in a separate folder to avoid having a large main executable file. I've tried several approaches, but I haven't been successful. I would appreciate any guidance on how to resolve this issue.

```
    public class ProgramPublishSingleFile
    {
        [STAThread]
        public static int Main(string[] args)
        {
            string rundir = Path.GetDirectoryName(System.Diagnostics.Process.GetCurrentProcess().MainModule!.FileName)!;
            var libraryLoader = new CefLibraryHandle(Path.Combine(rundir,"cefsharp", "libcef.dll"));
            var isValid = libraryLoader.IsInvalid;
            var exitCode = CefSharp.BrowserSubprocess.SelfHost.Main(args);
            if (exitCode >= 0)
            {
                return exitCode;
            }
            var settings = new CefSettings()
            {
                BrowserSubprocessPath = Path.Combine(rundir, "cefsharp", "CefSharp.BrowserSubprocess.exe"),
                LocalesDirPath = Path.Combine(rundir, "cefsharp", "locales"),
                ResourcesDirPath = rundir
            };
            Cef.Initialize(settings, performDependencyCheck: false);

            var app = new App();
            app.InitializeComponent();
            return app.Run();
        }
    }
```

**Comment by amaitland** [comment]
> Additionally, I would like to have the CEFSharp-related DLLs placed in a separate folder to avoid having a large main executable file.

I've never tested this. Get it working without doing this first.


> BrowserSubprocessPath = Path.Combine(rundir, "cefsharp", "CefSharp.BrowserSubprocess.exe"),

You must `self host` the browser sub process, you cannot use `CefSharp.BrowserSubprocess.exe`, you **must** use your own exe
